### ISSUE TITLE
Infinite loop for test case 9_12

### ISSUE THREAD

**Issue by crunchypotatoes** [nus-cs2030/2223-s2#838]
# Summary

I get an infinite loop for test case 9_12.

## Description:
This is pretty odd because it seems that others on who face infinite loops are at test case 11 or 15. Anyone else had this issue? How did you debug?

Given that I passed test case 11 suggests that it s not a matter of updating servers properly so Im not sure where is the issue :(

**Comment by ringleading** [comment]
ello, maybe a check on how your buffer/pending events are handled with the addition of your self-checkout counters?

**Comment by ringleading** [comment]
i had it log all the event's toString()s so i could pinpoint from the output where exactly went wrong. things like server avail time and which event was being repeated helped me in debugging mine

**Comment by crunchypotatoes** [comment]
will try thank you :)

**Comment by kyledilao777** [comment]
would be good to use print statements at all the points where you change the state of your queue and track if its consistent with how u designed it

**Comment by McFishMealUpsize** [comment]
This might have something to do with your intermediate classes calling themselves too many times. Either that or you are not accessing your self-checkout severs wrongly. You can try debugging this in the Simulator class by adding a for loop till "i" hits a certain number (I usually choose 100) and adjusting toString() methods.

**Comment by displeasedcat** [comment]
Depending on how you designed your code, you may have to amend the exit condition for your buffer according to the self checkout abit differently than how you did for human server

**Comment by territxl** [comment]
I think you might also consider if there are edge cases where a certain update of your queues will result in an infinite loop

**Comment by easypzlemonsqueezy** [comment]
yup it might be due to an edge case that may result in an infinite loop, this again depends on your implementation of your wait and possibly intermediate wait events
